Prog/Eviction Group/DC-evic-analysis-need.R

```diff
@@ -36,50 +34,92 @@ clean_pums <- dc_pums_22 %>%
     GRNTP = GRNTP*(ADJHSG/1000000), # adjusting to match years prior to 2022 to 2022 $ value for gross rent and household income 
     HINCP = HINCP*(ADJINC),
     inc_month_30 = (HINCP*0.30)/12, # 30% of monthly income 
+    inc_month_50 = (HINCP*0.50)/12, # 50% of monthly income
     rent_burden = case_when(
       inc_month_30 <= GRNTP ~ "Rent Burden", # If 30% of monthly income less than or equal to gross monthly rent cost, HH is rent burden 
       HINCP <= 0 & GRNTP > 0 ~ "Rent Burden", # If HH income is less than or equal to 0 and the HH pays rent, HH is rent burden
-      TRUE ~ "Not Rent Burden")) %>% # All else households not rent burden 
-  dmv_hh_inc() # to have HH income bins from function in DC-HH-inc-cat.R
+      TRUE ~ "Not Rent Burden"), # All else households not rent burden
+    severe_rent_burden = case_when(
+      inc_month_50 <= GRNTP ~ "Severe Rent Burden", 
+      HINCP <= 0 & GRNTP > 0 ~ "Severe Rent Burden", # If HH income is less than or equal to 0 and the HH pays rent, HH is rent burden
+      TRUE ~ "Not Severely Rent Burden")) %>%  
+  dmv_hh_inc() # attach HH income bins from function in DC-HH-inc-cat.R
 
```

Comment:
For me, this code generates a `clean_pums` dataset with 0 observations. I looked more into the `VACS` variable in `dc_pums_22`, and it looks like it takes the values "b" or 1-7. It doesn't look like VACS takes the value of 0, so I think the `filter()` function in line 29 creates a subset dataset with 0 observations. What did you mean by `VACS=="0"` here? Is it possible you recoded that variable in a different script?

Reply:
Here's the code I ran:

```
## Filter for only head of occupied households, create rent burden variable, and add 2022 income category bins
clean_pums <- dc_pums_22 %>%
  filter(VACS=="b", # not vacant #### TOMI CHANGED "0" to "b" here
         SPORDER == 1, # person 1 in household
         TEN == 3 | TEN == 4) %>% #renter
  mutate(across(-c(SERIALNO, VACS), as.numeric), # all numeric vars besides SERIALNO #### TOMI ADDED VACS TO KEEP AS CHARACTER
         year=substr(SERIALNO, 1, 4),
         GRNTP = GRNTP*(ADJHSG/1000000), # adjusting to match years prior to 2022 to 2022 $ value for gross rent and household income 
         HINCP = HINCP*(ADJINC),
         inc_month_30 = (HINCP*0.30)/12, # 30% of monthly income 
         inc_month_50 = (HINCP*0.50)/12, # 50% of monthly income
         rent_burden = case_when(
           inc_month_30 <= GRNTP ~ "Rent Burden", # If 30% of monthly income less than or equal to gross monthly rent cost, HH is rent burden 
           HINCP <= 0 & GRNTP > 0 ~ "Rent Burden", # If HH income is less than or equal to 0 and the HH pays rent, HH is rent burden
           TRUE ~ "Not Rent Burden"), # All else households not rent burden
         severe_rent_burden = case_when(
           inc_month_50 <= GRNTP ~ "Severe Rent Burden", 
           HINCP <= 0 & GRNTP > 0 ~ "Severe Rent Burden", # If HH income is less than or equal to 0 and the HH pays rent, HH is rent burden
           TRUE ~ "Not Severely Rent Burden")) %>%  
  dmv_hh_inc() # attach HH income bins from function in DC-HH-inc-cat.R

```

Reply:
Made a note on script that others may have to switch to "b"